libs/components/datetime/src/lib/modules/datepicker/calendar/daypicker-cell.component.spec.ts

```diff
@@ -67,146 +91,146 @@ describe('daypicker cell', () => {
     });
 
     fixture = TestBed.createComponent(SkyDayPickerCellComponent);
-    component = fixture.componentInstance;
-    component.date = {
-      current: false,
-      date: new Date(2021, 9, 1),
-      disabled: false,
-      label: '1',
-      secondary: false,
-      selected: false,
-      uid: '1',
-      keyDate: true,
-      keyDateText: ['important!'],
-    };
-    datepickerService = TestBed.inject(SkyDatepickerCalendarService);
+    componentRef = fixture.componentRef;
+    calendarSvc = TestBed.inject(SkyDatepickerCalendarService);
   });
 
   describe('set hasTooltip', () => {
     it('should set hasTooltip to false if not a key date', () => {
-      if (component.date) {
-        component.date.keyDate = false;
-      }
+      componentRef.setInput('date', {
+        ...CONTEXT_WITH_KEY_DATE,
+        keyDate: false,
+      } satisfies SkyDayPickerContext);
+
       fixture.detectChanges();
 
-      expect(component.hasTooltip).toBeFalsy();
+      expectHasPopoverTrigger(fixture, false);
     });
 
     it('should set hasTooltip to false if a key date and keyDateText undefined', () => {
-      if (component.date) {
-        component.date.keyDateText = undefined;
-      }
+      componentRef.setInput('date', {
+        ...CONTEXT_WITH_KEY_DATE,
+        keyDateText: undefined,
+      } satisfies SkyDayPickerContext);
 
       fixture.detectChanges();
 
-      expect(component.hasTooltip).toBeFalsy();
+      expectHasPopoverTrigger(fixture, false);
     });
 
     it('should set hasTooltip to false if a key date and no keyDateText items', () => {
-      if (component.date) {
-        component.date.keyDateText = [];
-      }
+      componentRef.setInput('date', {
+        ...CONTEXT_WITH_KEY_DATE,
+        keyDateText: [],
+      } satisfies SkyDayPickerContext);
 
       fixture.detectChanges();
 
-      expect(component.hasTooltip).toBeFalsy();
+      expectHasPopoverTrigger(fixture, false);
     });
 
     it('should set hasTooltip to false if a key date and keyDateText item empty', () => {
-      if (component.date) {
-        component.date.keyDateText = [''];
-      }
+      componentRef.setInput('date', {
+        ...CONTEXT_WITH_KEY_DATE,
+        keyDateText: [''],
+      } satisfies SkyDayPickerContext);
 
       fixture.detectChanges();
 
-      expect(component.hasTooltip).toBeFalsy();
+      expectHasPopoverTrigger(fixture, false);
     });
 
     it('should set hasTooltip to true if key date and keyDateText item not empty', () => {
+      componentRef.setInput('date', CONTEXT_WITH_KEY_DATE);
+
       fixture.detectChanges();
 
-      expect(component.hasTooltip).toBeTruthy();
+      expectHasPopoverTrigger(fixture, true);
+    });
+
+    it('should enable previously disabled tooltips when date value changes after init', () => {
```

Comment:
This test failed before making the necessary fixes in this branch. Essentially, the cell was "onceable" and didn't respond reactively to input changes.